test/test.js

```diff
@@ -890,340 +919,443 @@ describe('S3rver Tests', function () {
   });
 
   it('should reach the server with a bucket vhost', function (done) {
-    request({ url: 'http://localhost:4569/', headers: { host: buckets[0] + '.s3.amazonaws.com' } }, function (err, response, body) {
-      if (err) {
-        return done(err);
-      }
+    request({ url: s3Client.endpoint.href, headers: { host: buckets[0] + '.s3.amazonaws.com' } }, function (err, response, body) {
+      if (err) return done(err);
 
-      if (response.statusCode !== 200) {
-        return done(new Error('Invalid status: ' + response.statusCode));
-      }
+      response.statusCode.should.equal(200);
+      body.should.containEql('ListBucketResult');
+      done();
+    });
+  });
+});
 
-      if (body.indexOf('ListBucketResult') === -1) {
-        return done(new Error('Unexpected response: ' + body));
-      }
+describe('S3rver CORS Policy Tests', function () {
+  const bucket = 'foobars';
+  let s3Client;
 
-      done();
+  before('Initialize bucket', function (done) {
```

Comment:
Do you think we should add a cleanup step to this test suite? So as to not pollute the contents of the tmp directory for other suites (even if the others currently clean up the tmp directory before running)

Reply:
I'd say no, only because other tests should be cleaning up the directory they're using before running anyway, otherwise I'd consider that a form of dependence on other tests. Additionally we shouldn't have to be too concerned with the intermediate test files left behind in the tmpdir since the OS should clean them up on its own.

Reply:
I think tests should clean up after themselves, but we can revisit this in a future test cleanup.